### ISSUE TITLE
allow flow control Stages to have current Freight

### ISSUE THREAD

**Issue by krancour** [akuity/kargo#982]
See [this discussion](https://github.com/akuity/kargo/discussions/981).

Picture for reference:

![276237350-3243f302-a83d-4be2-b792-7f4715ebd819](https://github.com/akuity/kargo/assets/1821014/9f3397a5-17e7-414a-a419-928c56bfb286)

> "Control flow" Stages have no promotion mechanisms and aren't backed by an environment, and with this being the case, they never undergo any state changes. So it's not only that the UI isn't showing the control flow Stage's current Freight. The control flow Stage doesn't have any current Freight.
>
> With this being the case, the effect of the following buttons seems rather ambiguous:
>
>    * The buttons for pushing from code-testing and code-security to prod-ready
>    * The button for pulling to prod-ready from either of code-testing or code-security
>    * The button for pushing from prod-ready to prod-east and prod-west
>    * The buttons for pulling to prod-east or prod-west from prod-ready
>
> Probably this isn't just a UI fix. The most sensible thing may be to just start permitting control flow Stages to reference a current Freight even though no real state change will occur.

cc @jessesuen for thoughts.

**Comment by jessesuen** [comment]
Actually, it didn't make sense for me for control-flow stages to have "currentFreight". And when we [rename it](https://github.com/akuity/kargo/issues/842) to "lastPromotedFreight" it will make even less sense. And so it was intentional that the field was empty.

Theres still some work to implement more flow control logic in control flow stages (e.g. waitgroup like semantics). My preference would be to do nothing until we understand our plan for that.

**Comment by krancour** [comment]
> Actually, it didn't make sense for me for control-flow stages to have "currentFreight". And when we https://github.com/akuity/kargo/issues/842 to "lastPromotedFreight" it will make even less sense. And so it was intentional that the field was empty.

I agree with this.

I do remain confused about:

1. What the push/pull promote buttons do on and around the control flow stage remain ambiguous and possibly even broken.

1. After #631/#911, I think there's no way for Freight to get qualified for a control flow Stage, meaning Freight will stall there and not be accessible to downstream Stages. I'll open a separate issue about that.

> Theres still some work to implement more flow control logic in control flow stages

Yep. I think it's possible that such Stages are worthy of their own top-level type instead of trying to graft the logic for those onto the reconciler for normal Stages.

**Comment by krancour** [comment]
I'm closing this because I now agree with @jessesuen that these stages do not have current Freight by design and we should keep it that way.

In fact, changing that would promote more confusion in the vein of #1153
